### PR TITLE
[mergebot] rebase + merge

### PR DESCRIPTION
**Pull request by clee2000**
rebase and merge with one command (pytorchbot merge -r)

pytorch pr: https://github.com/pytorch/pytorch/pull/85028

### CODE REVIEW

File: torchci/lib/bot/pytorchBotHandler.ts [hunk 1]
```diff
@@ -176,12 +160,12 @@ The explanation needs to be clear on why this is needed. Here are some good exam
 
     if (!rejection_reason) {
       await this.logger.log("merge", extra_data);
-      await this.dispatchEvent(
-        "try-merge",
-        forceRequested,
-        mergeOnGreen,
-        landChecks || landChecksEnrolled
-      );
+      await this.dispatchEvent("try-merge", {
```
Comment (huydhn) [inline torchci/lib/bot/pytorchBotHandler.ts:163]: It's a nice tweak indeed to use the payload dictionary

File: torchci/lib/bot/cliParser.ts [hunk 2]
```diff
@@ -56,6 +56,10 @@ mergeOption.add_argument("-l", "--land-checks", {
   "reliability and decrease risk of revert. The tests added are: pull, Lint and trunk. Note " +
   "that periodic is excluded. (EXPERIMENTAL)",
 });
+mergeOption.add_argument("-r", "--rebase", {
```
Comment (huydhn) [inline torchci/lib/bot/cliParser.ts:59]: Should we accept branch name here, i.e. `-r viable/strict` to make this consistent in term of functionality with `@pytorchbot rebase -s|-b BRANCHNAME`.  The default branch could be set to `master`.

File: torchci/lib/bot/pytorchBotHandler.ts (hunk 1 above)
Comment (clee2000) [reply torchci/lib/bot/pytorchBotHandler.ts:163]: yea its pretty much https://github.com/pytorch/test-infra/pull/377 but i think that pr is a bit abandoned

File: torchci/lib/bot/cliParser.ts (hunk 2 above)
Comment (clee2000) [reply torchci/lib/bot/cliParser.ts:59]: done!

File: torchci/lib/bot/cliParser.ts [hunk 3]
```diff
@@ -56,6 +56,12 @@ mergeOption.add_argument("-l", "--land-checks", {
...
+  help: "Rebase the PR to re run checks before merging.  It will accept a branch name and " +
+  "will default to master if not specifiedd.",
```
Comment (janeyx99) [inline torchci/lib/bot/cliParser.ts:61]:
```suggestion
  "will default to master if not specified.",
```
nit, but also if we want to make this public to other repos, we probably want to handle the general default branch

**Review by janeyx99 — APPROVED** [review]

**Review by huydhn — APPROVED** [review]
Thank you for getting this done so quickly!

File: torchci/lib/bot/cliParser.ts (hunk 3 above)
Comment (clee2000) [reply torchci/lib/bot/cliParser.ts]: did the default branch thing

**Pull request merged**